src/circuits/qap.rs

```diff
@@ -15,19 +17,47 @@ pub enum CreationError {
 }
 
 impl Qap {
+    /// Creates a new QAP
+    /// This expects vectors to be organized like:
+    /// v0,w0,y0
+    /// inputs associated v,w,y polynomials
+    /// mid associated polynomials
+    /// outputs associated v,w,y polynomials
     pub fn new(
         v: Vec<Polynomial>,
         w: Vec<Polynomial>,
         y: Vec<Polynomial>,
         target: Polynomial,
+        amount_of_inputs: usize,
+        amount_of_outputs: usize,
```

Comment:
Can we name these `number_of_inputs` and `number_of_outputs`?

Reply:
Sure